server/src/routes/opportunities/index.ts

```diff
@@ -269,24 +345,38 @@ export const extendCV = (_: ILogger) => async (ctx: Router.RouterContext) => {
   setResponse(ctx, OK, Number(result.expires));
 };
 
-export const checkCVExistance = (_: ILogger) => async (ctx: Router.RouterContext) => {
-  const { githubId } = ctx.query;
+export const manageHiddenStatus = (_: ILogger) => async (ctx: Router.RouterContext) => {
+  const { githubId, isHidden: reqHiddenStatus } = ctx.request.body;
 
-  const user = await getRepository(CV).findOne({ where: { githubId } });
+  const cvRepository = getRepository(CV);
+  const cv = await cvRepository.findOne({ where: { githubId } });
+
+  if (cv === undefined) {
+    setResponse(ctx, NOT_FOUND);
+    return;
+  }
 
-  setResponse(ctx, OK, user !== undefined);
+  const prevHiddenStatus = cv.isHidden;
+
+  if (reqHiddenStatus === prevHiddenStatus) {
+    setResponse(ctx, OK, cv.isHidden);
+    return;
+  }
+
+  const result = await cvRepository.update({ githubId }, { isHidden: reqHiddenStatus });
+  setResponse(ctx, OK, result.raw.isHidden);
 };
 
 export function opportunitiesRoute(logger: ILogger) {
   const router = new Router<any, any>({ prefix: '/opportunities' });
-
-  router.get('/exists', guard, checkCVExistance(logger));
+  router.post('/hide', guard, manageHiddenStatus(logger));
   router.post('/extend', guard, extendCV(logger));
   router.get('/consent', guard, getOpportunitiesConsent(logger));
-  router.post('/consent', guard, setOpportunitiesConsent(logger));
-  router.get('/', guard, getJobSeekersData(logger));
-  router.get('/cv', guard, getCVData(logger));
-  router.put('/cv', guard, createCV(logger));
+  router.post('/consent', guard, manageOpportunitiesConsent(logger));
+  router.get('/jobseekers', guard, getVisibleJobSeekersData(logger));
+  router.get('/jobseekers-all', guard, getAllJobSeekersData(logger));
+  router.get('/cv-edit', guard, getEditCVData(logger));
```

Comment:
Because, unlike cv-edit, feedback and student data are added to the cv-view, there are additional request to db for it. I think I can replace these with one endpoint using query parameters